lib/ProductOpener/Tags.pm

```diff
@@ -1702,6 +1704,188 @@ sub build_tags_taxonomy($$$) {
 }
 
 
+=head2 generate_tags_taxonomy_extract ( $tagtype, $tags_ref, $options_ref, $lcs_ref)
+
+Generate an extract of the taxonomy for a specific set of tags.
+
+=head3 Parameters
+
+=head4 tag type $tagtype
+
+=head4 reference to a list of tags ids $tags_ref
+
+=head4 reference to a hash of key/value options
+
+Options:
+- fields: comma separated lists of fields (e.g. "name,description,vegan:en,inherited:vegetarian:en" )
+
+Properties can be requested with their name (e.g."description") or name + a specific language (e.g. "vegan:en").
+Only properties directly defined for the entry are returned.
+To include inherited properties from parents, prefix the property with "inherited:" (e.g. "inherited:vegan:en").
+
+- include_parents: include entries for all direct parents of the requested tags
+- include_children: include entries for all direct children of the requested tags
+
+=head4 reference to an array of language codes
+
+Languages for which we want to extract names, synonyms, properties.
+
+=cut
+
+sub generate_tags_taxonomy_extract ($$$$) {
+
+	my $tagtype = shift;
+	my $tags_ref = shift;
+	my $options_ref = shift;
+	my $lcs_ref = shift;
+
+	# Return empty hash if the taxonomy does not exist
+	if (not defined $translations_to{$tagtype}) {
+		return {};
+	}
+
+	# For the options include_children or include_parents,
+	# we will need to include data for more tags that requested.
+	# @tags will hold the tags to include
+
+	my @tags = ();
+	my %requested_tags = ();
+	foreach my $tagid (@$tags_ref) {
+		push @tags, $tagid;
+		$requested_tags{$tagid} = 1;
+	}
+
+	my $fields_ref;
+	my @properties = ();
+	if ((defined $options_ref) and (defined $options_ref->{fields})) {
+		$fields_ref = {};
+		foreach my $field (split(/,/, $options_ref->{fields}) ) {
+			# Compute a list of the requested inherited properties,
+			# as we will populate them directly
+			if ($field =~  /^inherited:(.*):(\w\w)$/) {
+				my $prop = $1;
+				my $lc = $2;
+				push @properties, [$prop, $lc];
+			}
+			# Compute a hash of the other requested fields
+			# as we will go through all existing properties
+			else {
+				$fields_ref->{$field} = 1;
+			}
+		}
+	}
+
+	my $taxonomy_ref = {};
+
+	while (my $tagid = shift @tags) {
+
+		if (defined $direct_parents{$tagtype}{$tagid}) {
+			
+			$taxonomy_ref->{$tagid}{parents} = [];
+			foreach my $parentid (sort keys %{$direct_parents{$tagtype}{$tagid}}) {
+				if ((not defined $fields_ref) or (defined $fields_ref->{parents})) {
```

Comment:
I don't understand the first part of the condition ! Does that mean we want all fields ?
In this case I would define it in a variable before : `my $all_fields = not defined $fields_ref` it would be easier to understand in condition : `if ($all_fields or (defined $fields_ref->{parents}) {`

Reply:
Good idea.